app/services/analytics/base_service.rb

```diff
@@ -1,15 +1,36 @@
 # frozen_string_literal: true
 
+require "lago_http_client"
+
 module Analytics
   class BaseService < BaseService
     def initialize(organization, **filters)
       @organization = organization
       @filters = filters
+
       super()
     end
 
     private
 
     attr_reader :organization, :filters, :records
+
+    def http_client
+      LagoHttpClient::Client.new(endpoint_url)
+    end
```

Comment:
Should we keep the client in memory to share it between request. So not in the service instance but somewhere.
Obviously, this kind of optimization can be done later.